core/src/main/java/nl/nn/adapterframework/validation/XSD.java

```diff
@@ -261,17 +264,32 @@ public int compareToByContents(XSD x) {
 	@Override
 	public abstract Reader getReader() throws IOException;
 
+	public static Set<IXSD> getXsdsRecursive(Set<IXSD> xsds) throws ConfigurationException {
+		return getXsdsRecursive(xsds, false);
+	}
 
-	@Override
-	public Set<IXSD> getXsdsRecursive(boolean supportRedifine) throws ConfigurationException {
-		return getXsdsRecursive(new HashSet<IXSD>(), supportRedifine);
+	public static Set<IXSD> getXsdsRecursive(Set<IXSD> xsds, boolean supportRedefine) throws ConfigurationException {
+		Map<String,IXSD> xsdsRecursive = new LinkedHashMap<>();
+		int i=0;
+		for (IXSD xsd : xsds) {
+			// All top level XSDs need to be added, with a unique systemId. If they come from a WSDL, they all have the same systemId.
+			String normalizedSystemId = FilenameUtils.normalize(xsd.getSystemId())+"["+(++i)+"]";
```

Comment:
What's the reason adding an index to the normalized system ID?

Reply:
That is to give each of the XSDs in the WSDL a unique systemID, to avoid collisions in the further processing

Reply:
Is there a test that covers this scenario?

What happens if one of the XSDs in the WSDL includes the other? Then it will be added twice, once with and once without index?

Reply:
- The normalized systemId is used as the key in the map. Without the index, all XSDs of the WSDL overwrite each other. The index was introduced to overcome this problem. You could consider that a test. 
- I don't think it is possible to refer to an XSD in a WSDL via its system id